### DISCUSSION TITLE
Just a Question from a beginner. | [openwrt]

### DISCUSSION THREAD

**Discussion by BitBlockv6** [x-cmd/x-cmd#200] [Q&A]
Hello, sorry i dont find a way to write my question on a other way. Google and ask Chatbot like Perplexity doesnt bring me a solution.
I installed x-cmd on my little server with debian 12 and it's all good and i'am still learning.
Then i had the idea to install x-cmd on my OpenWrt Router ( 2 Core armv7l ) the install was a success but when i want
to open for example the help ( x help ) i get an error message like that: "E|tui: The current screen size is insufficient to display the content"
I have to say that i connect with my Lenovo P11 Pro Gen2 Android Tablet with the App JuiceSSH but on Debian 12 its ok so i dont think that this is the 
failure. Perhaps it's the shell 'ash' that is standard on OpenWr ?! fish is functional on Openwrt. "x theme" for example opens normally. I have no idea where i can begin to search for the failure perhaps you have some tipps for me and sorry for my bad english.

Any help would be nice !

thank you for this nice program ....

**Comment by edwinjhlee** [comment]
Hi ~ Thanks for your support !

We don't have a test environment for openwrt. However, although we don't conduct massive tests of x-cmd on ash, it should work with ash by design.

## ui window size

> E|tui: The current screen size is insufficient to display the content

This log suggests the console windows size don't meet the least requirement of opening `x help` TUI app. 

Please try the command in the console windows where the command `x help` fail to run.

```bash
x ui col
x ui row
```

## openwrt

The pkg system may not works with openwrt. Because we currently only provide aarch64 binary build for linux.

**Comment by edwinjhlee** [comment]
It would be very helpful if the screenshot and logs and commands provided.

**Comment by BitBlockv6** [comment]
Hello, i'am very sorry that i not answered in time.
I managed to get the most commands i found very interesting on the first sight for example the '@gemini' command worked with setting the COLUMNS and LINES to a higher variable.
Sadly sometimes something errors to install sometimes not but its not usable for example i tried to bring java on OpenWrt it installs but the java binary is not usable. 
But its not worth the work to get behind this like you said its intended for aarch64 not the old armv7l architecture.

Again thank you for your great work ! Keep it on ( perhaps on Termux xD ) !

**Comment by edwinjhlee** [comment]
> I managed to get the most commands i found very interesting on the first sight for example the 'https://github.com/gemini' command worked with setting the COLUMNS and LINES to a higher variable.

Cool ~ Yes. The tui depends on COLUMNS and LINES. Bormally x-cmd could detect it and set COLUMNS and LINES correctly even in ash, and the TUI app could run normally, provided that if your current terminal is big enough.


> Sadly sometimes something errors to install sometimes not but its not usable for example i tried to bring java on OpenWrt it installs but the java binary is not usable.
> But its not worth the work to get behind this like you said its intended for aarch64 not the old armv7l architecture.

It worths the work. X-CMD should works on armv7l which is widely used in some energy-efficient embedding hardwares. I am currently just swamped by other issues related to the making x-cmd finally stable.

**Comment by BitBlockv6** [comment]
Hello,
i have an issue and dont know where it comes from.
when i use x co or other commands i become this error: 

UI select can only be used in manual mode

What can I do ?

**Reply by Longlone29** [reply]
> Hello, i have an issue and dont know where it comes from. when i use x co or other commands i become this error:
> 
> UI select can only be used in manual mode
> 
> What can I do ?

meet same problem in `x install`, need fix

**Comment by edwinjhlee** [comment]
This might be a bug in recent versions.  
We implemented a more cautious approach to prevent autonomous scripting from hanging. It seems it might be triggered unexpectedly.

An update with a more approprivate implementation will be released within the next 10 days.

**Comment by edwinjhlee** [comment]
I am moving this issue to discussion ~ And we welcome more feedback and questions ~
